ynr/apps/sopn_parsing/helpers/extract_tables.py

```diff
@@ -19,19 +19,25 @@ def extract_ballot_table(ballot, parse_flavor="lattice"):
         raise ValueError(
             "Pages for table not known for document, extract page numbers first"
         )
-
     try:
         tables = camelot.read_pdf(
             document.uploaded_file.path,
             pages=document.relevant_pages,
-            flavor=parse_flavor,
+            flavor="lattice",
         )
     except (NotImplementedError, AttributeError):
         # * NotImplementedError is thrown if the PDF is an image or generally
         #   unreadable.
         # * AttributeError is thrown on some PDFs saying they need a password.
         #   Assume this is a bug in camelot, and ignore these PDFs
-        raise NoTextInDocumentError()
+        try:
```

Comment:
The way we have this written, it never hits the second `try` block with `stream` and moves right down to L44 after the first `try` block even though it returns empty tables.  I reversed the order and tried `stream` first and the table was found in the case of `local.bradford.baildon.2021-05-06`. How can we refactor within the `except` - or should we move the second try block to L50 before we try to `return`?

Reply:
If I understand rightly, is this because lattice is throwing an `NoTextInDocumentError`?

If so, you should be able to add that to the list of exceptions caught on line 28/29 above, but I expect stream will throw the same error, so it might not solve the actual problem?

Reply:
Changing the parsing flavour doesn't solve the parsing issue in the case of `local.bradford.baildon.2021-05-06` but it does help move it along to the next bug in parsing tables.

Reply:
@VirginiaDooley yes you are right, I encountered the same thing. It is because neither a NotImplementedError or AttributeError are raised by the `read_pdf` method for these set of ballots. Which I guess is a good thing, as it means that we are always able to parse some tables where we have pages?

One other thing I noticed is that wherever we get a "Skipping... due to parse error" this is because we failed to extract the page numbers in the first stage of parsing. So we throw a `ValueError` [on line 19](https://github.com/DemocracyClub/yournextrepresentative/pull/1694/commits/d840bae7db560fccb082889f1db75f9e78e520b8#diff-cb93e9475f5c03e3d9bf7d1834371e4da4097ee2c1a434d8b33f53e5c67a5943R19). Which I got caught out by, so I am working on improving the error handling to make it a bit more explicit about where failures are occurring